bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/AbstractDiscoveryService.java

```diff
@@ -248,6 +248,7 @@ protected synchronized void stopScan() {
      *            Holds the information needed to identify the discovered device.
      */
     protected void thingDiscovered(DiscoveryResult discoveryResult) {
+        discoveryResult.discoveredBy(this.getClass().getName());
```

Comment:
Is this really the right place to set this property? After all, the DiscoveryResult interface defines it and thus potential implementors probably feel obliged to fill it (before passing it here). I think at this point, the passed instance should be treated as an immutable (otherwise you shouldn't have added the property to the interface at all, but only to our internal implementation that the builder creates).
There should at least be some safe guard to check whether it is already set (and then do nothing) and if it is not set, try to set it, but catch exceptions.